docs/dotnet/dotnet-aspnet-metrics.md

```diff
@@ -0,0 +1,216 @@
+# Semantic Conventions for ASP.NEt Core specific metrics
+
+**Status**: [Experimental][DocumentStatus]
+
+This document defines semantic conventions for ASP.NET Core metrics, not specific to HTTP
+
+**Disclaimer:** These are initial .NET metric instruments available in .NET 8 but more may be added in the future.
+
+<!-- toc -->
+
+- [Routing](#routing)
+  * [Metric: `aspnet.routing.successful_matches`](#metric-aspnetroutingsuccessful_matches)
+  * [Metric: `aspnet.routing.failed_matches`](#metric-aspnetroutingfailed_matches)
+- [Exception metrics](#exception-metrics)
+  * [Metric: `aspnet.diagnostics_handler.exceptions`](#metric-aspnetdiagnostics_handlerexceptions)
+- [Rate-limiting](#rate-limiting)
+  * [Metric: `aspnet.rate_limiting.active_request_leases`](#metric-aspnetrate_limitingactive_request_leases)
+  * [Metric: `aspnet.rate_limiting.request_lease.duration`](#metric-aspnetrate_limitingrequest_leaseduration)
+  * [Metric: `aspnet.rate_limiting.queued_requests`](#metric-aspnetrate_limitingqueued_requests)
+  * [Metric: `aspnet.rate_limiting.queued_requests.duration`](#metric-aspnetrate_limitingqueued_requestsduration)
+  * [Metric: `aspnet.rate_limiting.rejected_requests`](#metric-aspnetrate_limitingrejected_requests)
+
+<!-- tocstop -->
+
+## Routing
+
+All routing metrics are reported by `Microsoft.AspNetCore.Routing` meter.
+
+### Metric: `aspnet.routing.successful_matches`
+
+**Status**: [Experimental][DocumentStatus]
+
+**TODO: can we try one aspnet.routing.matches** ?
+
+This metric is required.
+
+<!-- semconv metric.aspnet.routing.successful_matches(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `aspnet.routing.successful_matches` | Counter | `{match}` | Number of requests that successfully matched to an endpoint. [1] |
```

Comment:
Porting active comment threads from the word doc:

## Liudmila Molkova
why not ....routing.matches with status dimension? success/failure might not be enough and it's common to have 1 metric with extendable attribute representing status

July 18, 2023 at 4:05 PM
## James Newton-King
I can't remember why I decided to have two counters here. Maybe because the route and fallback tags only made sense if the match was successful? But they could be left off for a failed match...

July 18, 2023 at 8:50 PM
## Noah Falk
I take it you think merging it would be fine James? It makes sense to me.

July 19, 2023 at 4:30 AM

Reply:
Updated the PR and merged counters. We can get back to it once James is back.